File: eventbus/app/src/main/java/com/example/eventbus/EventBusApp.java
package com.example.eventbus;

import android.app.Application;

import org.greenrobot.eventbus.EventBus;

public class EventBusApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        EventBus.builder()
                // have a look at the index class to see which methods are picked up
                // if not in the index @Subscribe methods will be looked up at runtime (expensive)
                .addIndex(new MyEventBusIndex())
                .installDefaultEventBus();
    }
}

File: eventbus-kotlin/app/src/main/java/com/example/eventbuskotlin/EventBusApp.kt
package com.example.eventbuskotlin

import android.app.Application
import org.greenrobot.eventbus.EventBus

class EventBusApp: Application() {

    override fun onCreate() {
        super.onCreate()

        EventBus.builder()
                // have a look at the index class to see which methods are picked up
                // if not in the index @Subscribe methods will be looked up at runtime (expensive)
                .addIndex(MyEventBusIndex())
                .installDefaultEventBus()
    }

}